pkg/migrate/cstor/pool.go

```diff
@@ -178,6 +191,45 @@ func (c *CSPCMigrator) migrate(spcName string) error {
 	return nil
 }
 
+func (c *CSPCMigrator) checkForExistingCSPC(spcName string) error {
+	spcObj, err := spc.NewKubeClient().Get(spcName, metav1.GetOptions{})
+	if err == nil {
+		if spcObj.Annotations == nil || spcObj.Annotations[types.CStorPoolClusterLabelKey] == "" {
+			if spcObj.Annotations == nil {
+				spcObj.Annotations = map[string]string{}
+			}
+			spcObj.Annotations[types.CStorPoolClusterLabelKey] = c.CSPCName
+			_, err := spc.NewKubeClient().Update(spcObj)
+			if err != nil {
+				return err
+			}
+		} else {
+			if spcObj.Annotations[types.CStorPoolClusterLabelKey] != c.CSPCName {
+				return errors.Errorf("invalid cspc-name: the spc %s already set to be renamed as %s",
+					spcName,
+					spcObj.Annotations[types.CStorPoolClusterLabelKey],
+				)
+			}
+		}
+	}
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+	cspc, err := c.OpenebsClientset.CstorV1().
+		CStorPoolClusters(c.OpenebsNamespace).
+		Get(c.CSPCName, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		return nil
+	}
+	if err == nil {
```

Comment:
Now I have another concern with configuration if the user does in below manner:
- The user has multiple SPCs in his setup and launched a job(By applying job YAML by setting `cspc-name` argument) to migrate to CSPC(Consider pool is successfully migrated).
- He reused the same Job YAML to migrate another SPC but now somehow he didn't change the `cspc-name` with this steps things will wrong(I can see job will update the blockdeviceclaim labels with `cspc-name` argument and return error).

Reply:
We can overcome this by adding old spc name on the cspc and verify that it isn't from another migrated spc.

Reply:
Done

Reply:
Still, this problem exists but now it will patch SPC and fail(Can we prevent updating SPC annotation itself?)  @shubham14bajpai